doc/release/upcoming_changes/12065.performance.rst

```diff
@@ -0,0 +1,8 @@
+Faster version of ``np.isin`` and ``np.in1d`` for integer arrays
+----------------------------------------------------------------
+``np.in1d`` (used by ``np.isin``) can now switch to a faster algorithm
+(up to >10x faster) when it is fed two integer arrays.
+The algorithm bares similarities to a counting sort in that it
+uses the ``test_elements`` argument to index a boolean helper
```

Comment:
The `test_elements` and `elements` kwargs are defined on `isin()`, but this paragraph technically talks about `in1d()`. I don't know if this can be confusing for anybody (i.e. that there is no `test_elements` parameter for `in1d()`, because it calls its parameters `ar1` and `ar2`).

Reply:
Good point. I cleaned this up and also mentioned some things about the memory allocation being different.